types/gatherer.d.ts

```diff
@@ -55,28 +58,52 @@ declare global {
       trace?: Trace;
     }
 
-    export type PhaseResultNonPromise = void|LH.GathererArtifacts[keyof LH.GathererArtifacts]
+    export type PhaseArtifact = LH.GathererArtifacts[keyof LH.GathererArtifacts]
+    export type PhaseResultNonPromise = void|PhaseArtifact
     export type PhaseResult = PhaseResultNonPromise | Promise<PhaseResultNonPromise>
 
     export type GatherMode = 'snapshot'|'timespan'|'navigation';
 
-    export interface GathererMeta {
+    export type DefaultDependenciesKey = '__none__'
+    export type DependencyKey = keyof GathererArtifacts | DefaultDependenciesKey
+
+    interface GathererMetaNoDependencies {
+      /**
+       * Used to validate the dependency requirements of gatherers.
+       * If this property is not defined, this gatherer cannot be the dependency of another. */
+      symbol?: Symbol;
```

Comment:
Any reason to make this symbol optional?

Reply:
I was thinking "If this property is not defined, this gatherer cannot be the dependency of another." was the feature, to prevent unintentional dependencies and keep the graph relatively small. 

Also a bit of a hassle to require a property to be defined that won't be used in most cases.

Reply:
Why a symbol? Could `Class` or `Class.prototype` work?

Reply:
Maybe I'm misunderstanding, but what purpose would configuration serve (seems like it might not be possible?) if the dependency is hard-coded to a specific implementation?